js/todoapp.js

```diff
@@ -0,0 +1,184 @@
+  },
+];
+
+function todosVisibility(
+  visibility,
+  display1,
+  display2,
+  isShowAll,
+  isShowActive
+) {
+  trashCans.forEach((e) => (e.style.visibility = visibility));
+  deleteCompletedDiv.style.display = display1;
+  taskAddDiv.style.display = display2;
+  for (let todo of allTodos) {
+    if (isShowAll) {
+      todo.style.display = "block";
+    } else if (isShowActive) {
+      todo.style.display =
+        todo.getAttribute("completed") === "true" ? "none" : "block";
+    } else {
+      todo.style.display =
+        todo.getAttribute("completed") === "true" ? "block" : "none";
+    }
+  }
+}
+
+//**Todo creation**
+const todos = document.createElement("ul");
+todos.className = "todo-ul";
+function addTodo(todo) {
+  const todosDiv = document.querySelector("#tab1");
+  const newTodo = document.createElement("li");
+  newTodo.className = "todo-ul-item";
+  newTodo.setAttribute("completed", todo.completed);
+  newTodo.innerHTML = `
+            <input id="${todo.id}" type="checkbox" class="check">
+            <label class="task-label" for="${todo.id}">${todo.name}</label>
+            <button class="completed-trash">
+            <i class="fa-solid fa-trash-can"></i>
+            </button>
+    `;
+  todos.appendChild(newTodo);
+  todosDiv.appendChild(todos);
+}
+
+// **using demo todo**
+let tasks = storage.get("tasks") ?? demoTodo;
+function displayTodos() {
+  for (let task of tasks) {
+    addTodo(task);
+  }
+}
+
+displayTodos();
+
+// **Creating and adding todo**
+const input = document.querySelector(".task-input");
+const addButton = document.querySelector(".add-button");
+addButton.addEventListener("click", () => {
+  if (input.value) {
+    const randomID = Math.random();
+    const userInput = {
+      name: input.value,
+      id: randomID,
+      completed: false,
+    };
+
+    tasks.push(userInput);
+    storage.set("tasks", tasks);
+
+    addTodo(userInput);
+    input.value = "";
+
+    allCheckBoxes = document.querySelectorAll(".check");
+    completeTodo();
+  }
+});
+
+let allCheckBoxes = document.querySelectorAll(".check");
+function completeTodo() {
+  for (let checkbox of allCheckBoxes) {
+    const currentTodo = checkbox.parentElement;
+    const label = currentTodo.children[1];
+    if (currentTodo.getAttribute("completed") === "true") {
+      checkbox.checked = true;
+      label.style.textDecoration = "line-through";
+    } else {
+      checkbox.checked = false;
+    }
+
+    checkbox.addEventListener("click", () => {
+      const currentTodoId = Number(currentTodo.firstElementChild.id);
+      currentTodo.setAttribute("completed", checkbox.checked);
+      checkbox.checked
+        ? (label.style.textDecoration = "line-through")
+        : (label.style.textDecoration = "none");
+
+      for (let task of tasks) {
+        if (task.id === currentTodoId) {
+          task.completed = checkbox.checked;
+        }
+      }
+      storage.set("tasks", tasks);
+    });
+  }
+}
+
+completeTodo();
```

Comment:
`completeTodo` is a strange name for this function

Complete todo means "complete one single todo item" 

The logic of this function is to: render correct state for a todo and initialize event listeners for it. The valid name is `initializeTodos` or `setupTodos`, but no `complete`.